site/src/pages/TemplateVersionEditorPage/TemplateVersionEditorPage.tsx

```diff
@@ -1,115 +1,359 @@
-          missingVariables={editorState.context.missingVariables}
-          onSubmitMissingVariableValues={(values) => {
-            sendEvent({
-              type: "SET_MISSING_VARIABLE_VALUES",
-              values,
+          resources={resourcesQuery.data}
+          buildLogs={logs}
+          isPromptingMissingVariables={isMissingVariablesDialogOpen}
+          missingVariables={missingVariables}
+          onSubmitMissingVariableValues={async (values) => {
+            if (!uploadFileMutation.data) {
+              return;
+            }
+            onBuildStart();
+            const newVersion = await createTemplateVersionMutation.mutateAsync({
+              provisioner: "terraform",
+              storage_method: "file",
+              tags: {},
+              template_id: templateQuery.data.id,
+              file_id: uploadFileMutation.data.hash,
+              user_variable_values: values,
             });
+            onBuildEnds(newVersion);
+            setIsMissingVariablesDialogOpen(false);
           }}
           onCancelSubmitMissingVariableValues={() => {
-            sendEvent({
-              type: "CANCEL_MISSING_VARIABLE_VALUES",
-            });
+            setIsMissingVariablesDialogOpen(false);
           }}
         />
       )}
     </>
   );
 };
 
+const useFileTree = (templateVersion: TemplateVersion | undefined) => {
+  const tarFileRef = useRef<TarReader | null>(null);
+  const fileQuery = useQuery({
+    ...file(templateVersion?.job.file_id ?? ""),
+    enabled: templateVersion !== undefined,
+  });
+  const [fileTree, setFileTree] = useState<FileTree>();
+  useEffect(() => {
+    const initializeFileTree = async (file: ArrayBuffer) => {
+      const tarReader = new TarReader();
+      await tarReader.readFile(file);
+      tarFileRef.current = tarReader;
+      const fileTree = await createTemplateVersionFileTree(tarReader);
+      setFileTree(fileTree);
+    };
+
+    if (fileQuery.data) {
+      initializeFileTree(fileQuery.data).catch(() => {
+        console.error("Error on initializing the editor");
+      });
+    }
+  }, [fileQuery.data]);
+
+  return {
+    fileTree,
+    tarFile: tarFileRef.current,
+  };
+};
+
+const useVersionLogs = (
+  templateVersion: TemplateVersion | undefined,
+  options: { onDone: () => Promise<unknown> },
+) => {
+  const [logs, setLogs] = useState<ProvisionerJobLog[]>();
+  const templateVersionId = templateVersion?.id;
+  const refetchTemplateVersion = options.onDone;
+  const templateVersionStatus = templateVersion?.job.status;
+
+  useEffect(() => {
+    if (!templateVersionId || !templateVersionStatus) {
+      return;
+    }
+
+    if (templateVersionStatus !== "running") {
+      return;
+    }
+
+    const socket = watchBuildLogsByTemplateVersionId(templateVersionId, {
+      onMessage: (log) => {
+        setLogs((logs) => (logs ? [...logs, log] : [log]));
+      },
+      onDone: async () => {
+        await refetchTemplateVersion();
```

Comment:
Could the `await` be removed, since there's only one line of code here?

Reply:
Yes.